### ISSUE TITLE
ContrastChecker does not work in site editor

### ISSUE THREAD

**Issue by nielslange** [WordPress/gutenberg#62213]
## What problem does this address?

When editing colors in the page editor, the color contrast is checked by the ContrastChecker:

<img width="854" alt="Screenshot 2024-06-03 at 11 21 56" src="https://github.com/WordPress/gutenberg/assets/3323310/5f272a58-a622-4621-9868-3c46cddf1633">

However, in the site editor, the color contrast is not checked:

<img width="851" alt="Screenshot 2024-06-03 at 11 25 46" src="https://github.com/WordPress/gutenberg/assets/3323310/48d28f21-345a-4846-9d9a-d4be94fdc23a">

## What is your proposed solution?

I propose implementing the color contrast check in the site editor, just as it is done in the page editor.
